### PR TITLE
login and signup pages added, some bugs with makeshift endpoints

### PR DESCRIPTION
**Pull request by MaxGunter99**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "react-dom": "^16.8.6",
+    "react-popper": "^1.3.3",
+    "react-redux": "^7.0.2",
+    "react-router-dom": "^5.0.0",
+    "react-scripts": "2.1.8",
+    "react-transition-group": "^4.0.0",
+    "reactstrap": "^8.0.0",
+    "redux": "^4.0.1",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.3.0"
```
Comment (VeraSimon) [inline frontend/package.json:19]: Please move redux-thunk to dev dependencies. You shouldn't be deploying it to production.

File: frontend/src/Actions/LoginAndSignup.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export const SIGNUP_START = "SIGNUP_START";
+export const SIGNUP_SUCCESS = "SIGNUP_SUCCESS";
+export const SIGNUP_ERROR = "SIGNUP_ERROR";
+
+//action creators - login for login component, getTacos for CDM within protected home route
+
+export const login = creds => dispatch => {
+  dispatch({ type: LOGIN_START });
+
+  const endpoint = "https://dev-desk-info.herokuapp.com/api/login";
```
Comment (VeraSimon) [inline frontend/src/Actions/LoginAndSignup.js:18]: Might I suggest something like 
```
export const apiBaseUrl =
	process.env.NODE_ENV === 'production'
		? 'https://anywhere-reader-test.herokuapp.com'
		: 'http://localhost:8000';
```
?

File: frontend/src/Components/Login.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            name="username"
+            type="text"
+            placeholder="username"
+            value={this.state.credentials.username}
+            onChange={this.handleChanges}
+          />
+          <br />
+          <input
+            name="password"
+            type="text"
```
Comment (VeraSimon) [inline frontend/src/Components/Login.js:42]: Consider changing this type to `password` to obscure the field

File: frontend/src/Components/PrivateRoute.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { Route, Redirect, withRouter } from "react-router-dom";
+import { connect } from 'react-redux'
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+  return (
+    <Route
+      {...rest}
+      render={props =>
+        localStorage.getItem("jwt") ? (
```
Comment (VeraSimon) [inline frontend/src/Components/PrivateRoute.js:10]: Not as big of a deal here, but with larger projects, you'll actually want to confirm the JWT belongs to who you think it belongs to before grabbing any local data for them.

File: frontend/src/Components/Signup.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            name="username"
+            type="text"
+            placeholder="username"
+            value={this.state.credentials.username}
+            onChange={this.handleChanges}
+          />
+          <br />
+          <input
+            name="password"
+            type="text"
```
Comment (VeraSimon) [inline frontend/src/Components/Signup.js:42]: Consider updating this to a `password` field type.

**Comment by MaxGunter99** [comment]
will do, i was experimenting with a local host server, thats why i have so
many dependancies there

File: package.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (VeraSimon) [inline package.json:1]: You seem to have 2 package.json files OwO

File: yarn.lock [hunk 7]
```diff
@@ -0,0 +1,445 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (VeraSimon) [inline yarn.lock:1]: You seem to have 2 yarn.lock files OwO

**Review by VeraSimon — APPROVED** [review]

**Pull request merged**